File: Modules/Common/Resources/SIE.Wpf.Resources/CalendarSchemes/Commands/CalendarSchemeWeekDeleteCommand.cs
using SIE.Resources.CalendarSchemes;
using SIE.Wpf.Command;
using System;

namespace SIE.Wpf.Resources.CalendarSchemes.Commands
{
    /// <summary>
    /// 周日历删除命令
    /// </summary>
    [Command(ImageName = "DeleteEntity", Label = "删除", GroupType = CommandGroupType.Edit)]
    public class CalendarSchemeWeekDeleteCommand : ListDeleteCommand
    {
        /// <summary>
        /// 是否可执行
        /// </summary>
        /// <param name="view">视图</param>
        /// <returns>bool</returns>
        public override bool CanExecute(ListLogicalView view)
        {
            var calendarSchemeWeek = view.Current as CalendarSchemeWeek;
            if (calendarSchemeWeek == null || calendarSchemeWeek.ActiveDate <= DateTime.Today
                || (view.Data.Count == 1 && (view.Parent.Current as CalendarScheme).IsEnable == YesNo.Yes))
            {
                return false;
            }

            return true;
        }
    }
}

File: Modules/Common/Resources/SIE.Web.Resources/CalendarSchemes/Commands/CalendarSchemeWeekDeleteCommand.js
SIE.defineCommand('SIE.Web.Resources.CalendarSchemes.Commands.CalendarSchemeWeekDeleteCommand', {
    extend: 'SIE.cmd.Delete',
    meta: { text: "删除", group: "edit" },
    
    canExecute: function (view) {
        this.selectedItems = view.getSelection();
        if (this.selectedItems.length > 1) {
            return false;
        }
        var entity = view.getCurrent();
        if (entity == null || entity.data.ActiveDate <= new Date() || (view.getData().data.items.length == 1 && view.getParent().getCurrent().data.IsEnable == 1)) {
            return false;
        }
        return true;
    }
});